# IDX Exchange Data Analysis Intern Program 
(Week 2-3) EDA for the Feb 2026 Listings & Sold created CSVs 
- Part 1 : Listings

In [2]:
import pandas as pd
import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

In [30]:
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format', '{:.0f}'.format)

In [4]:
listings = pd.read_csv("CRMLSListing202602.csv")

EDA 'Listings'

In [5]:
listings.shape

(22058, 82)

In [6]:
listings.columns

Index(['OriginalListPrice', 'ListingKey', 'ListAgentEmail', 'CloseDate',
       'ClosePrice', 'ListAgentFirstName', 'ListAgentLastName', 'Latitude',
       'Longitude', 'UnparsedAddress', 'PropertyType', 'LivingArea',
       'ListPrice', 'DaysOnMarket', 'ListOfficeName', 'BuyerOfficeName',
       'CoListOfficeName', 'ListAgentFullName', 'CoListAgentFirstName',
       'CoListAgentLastName', 'BuyerAgentMlsId', 'BuyerAgentFirstName',
       'BuyerAgentLastName', 'FireplacesTotal', 'AssociationFeeFrequency',
       'AboveGradeFinishedArea', 'ListingKeyNumeric', 'MLSAreaMajor',
       'TaxAnnualAmount', 'CountyOrParish', 'PropertyType.1', 'MlsStatus',
       'ElementarySchool', 'ListAgentFirstName.1', 'AttachedGarageYN',
       'ParkingTotal', 'BuilderName', 'PropertySubType', 'LotSizeAcres',
       'SubdivisionName', 'BuyerOfficeAOR', 'YearBuilt', 'DaysOnMarket.1',
       'StreetNumberNumeric', 'LivingArea.1', 'ListingId',
       'BathroomsTotalInteger', 'City', 'TaxYear', 'BuildingAreaTot

In [7]:
listings.tail(5)

,OriginalListPrice,ListingKey,ListAgentEmail,CloseDate,ClosePrice,ListAgentFirstName,ListAgentLastName,Latitude,Longitude,UnparsedAddress,PropertyType,LivingArea,ListPrice,DaysOnMarket,ListOfficeName,BuyerOfficeName,CoListOfficeName,ListAgentFullName,CoListAgentFirstName,CoListAgentLastName,BuyerAgentMlsId,BuyerAgentFirstName,BuyerAgentLastName,FireplacesTotal,AssociationFeeFrequency,AboveGradeFinishedArea,ListingKeyNumeric,MLSAreaMajor,TaxAnnualAmount,CountyOrParish,PropertyType.1,MlsStatus,ElementarySchool,ListAgentFirstName.1,AttachedGarageYN,ParkingTotal,BuilderName,PropertySubType,LotSizeAcres,SubdivisionName,BuyerOfficeAOR,YearBuilt,DaysOnMarket.1,StreetNumberNumeric,LivingArea.1,ListingId,BathroomsTotalInteger,City,TaxYear,BuildingAreaTotal,BedroomsTotal,ContractStatusChangeDate,Longitude.1,ElementarySchoolDistrict,CoBuyerAgentFirstName,PurchaseContractDate,ListingContractDate,BelowGradeFinishedArea,BusinessType,Latitude.1,ListPrice.1,StateOrProvince,CoveredSpaces,MiddleOrJuniorSchool,FireplaceYN,Stories,HighSchool,Levels,ListAgentLastName.1,CloseDate.1,LotSizeDimensions,LotSizeArea,MainLevelBedrooms,NewConstructionYN,GarageSpaces,HighSchoolDistrict,PostalCode,BuyerOfficeName.1,AssociationFee,LotSizeSquareFeet,MiddleOrJuniorSchoolDistrict,UnparsedAddress.1
22053,1349900.0,1152811470,Pam@TeamEtem.com,2026-03-16,1320000.0,Pamela,Etem,33.717782,-118.020937,6172 Shields,Residential,1406.0,1349900.0,14,The OC Real Estate,"Wynn Real Estate, Inc.",The OC Real Estate,Pamela Etem,Nigel,Etem,OCTRASTEV,Steven,Tran,NaN,NaN,NaN,1152811470,17 - Northwest Huntington Beach,NaN,Orange,Residential,Closed,NaN,Pamela,True,7.0,Prestige,SingleFamilyResidence,0.1377,Prestige (17) Tr 6181 (PRES6),OrangeCounty,1965.0,14,6172.0,1406.0,OC26030438,2.0,Huntington Beach,NaN,NaN,3.0,2026-03-16,-118.020937,NaN,NaN,2026-02-24,2026-02-09,NaN,NaN,33.717782,1349900.0,CA,NaN,NaN,True,1.0,Ocean View,One,Etem,2026-03-16,60 x 100,6000.00,3.0,False,2.0,Huntington Beach Union High,92647,"Wynn Real Estate, Inc.",0.0,6000.0,NaN,6172 Shields
22054,3100.0,1152811444,sarainrealestate@yahoo.com,2026-03-10,3100.0,Sara,Ramirez,34.118764,-118.156722,600 Meridian Avenue,ResidentialLease,969.0,3100.0,24,M&M Bryant Real Estate Inc,M&M Bryant Real Estate Inc,NaN,Sara Ramirez,NaN,NaN,P22275,Sara,Ramirez,NaN,NaN,NaN,1152811444,658 - So. Pasadena,NaN,Los Angeles,ResidentialLease,Closed,NaN,Sara,False,2.0,NaN,SingleFamilyResidence,0.0859,Not Applicable,PasadenaFoothills,1948.0,24,600.0,969.0,P1-25814,1.0,South Pasadena,NaN,NaN,2.0,2026-03-10,-118.156722,NaN,NaN,2026-03-06,2026-02-09,NaN,NaN,34.118764,3100.0,CA,NaN,NaN,False,1.0,NaN,One,Ramirez,2026-03-10,NaN,3741.00,NaN,NaN,2.0,NaN,91030,M&M Bryant Real Estate Inc,0.0,3741.0,NaN,600 Meridian Avenue
22055,465000.0,1152811435,sheri@sakkrealty.com,2026-03-20,485000.0,Sheri,Jones,33.773338,-116.933692,1584 Napoli Way,Residential,1397.0,465000.0,7,S.A.K.K Realty,"HomeSmart, Evergreen Realty",S.A.K.K Realty,Sheri Jones,Karen,Belcher,PGALJAV,Javier,Galvez,NaN,NaN,NaN,1152811435,SRCAR - Southwest Riverside County,NaN,Riverside,Residential,Closed,NaN,Sheri,True,2.0,NaN,SingleFamilyResidence,0.2200,NaN,PacificWest,2001.0,7,1584.0,1397.0,PTP2601082,2.0,San Jacinto,NaN,NaN,3.0,2026-03-20,-116.933692,NaN,NaN,2026-02-18,2026-02-10,NaN,NaN,33.773338,465000.0,CA,NaN,NaN,True,1.0,NaN,One,Jones,2026-03-20,NaN,0.22,NaN,False,2.0,San Jacinto Unified,92583,"HomeSmart, Evergreen Realty",0.0,9583.2,NaN,1584 Napoli Way
22056,599900.0,1152811419,alvinmullinsbroker@gmail.com,NaN,NaN,Alvin,Mullins,33.953730,-117.405289,4693 Central,Residential,1776.0,599900.0,27,Your Home Team Real Estate,NaN,NaN,Alvin Mullins,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1152811419,252 - Riverside,NaN,Riverside,Residential,Active,NaN,Alvin,False,4.0,NaN,SingleFamilyResidence,0.1400,NaN,NaN,1950.0,27,4693.0,1776.0,RS26030176,2.0,Riverside,NaN,NaN,3.0,2026-02-09,-117.405289,NaN,NaN,NaN,2026-02-09,NaN,NaN,33.953730,599900.0,CA,NaN,NaN,True,1.0,NaN,One,Mullins,NaN,NaN,6098.00,3.0,False,2

In [33]:
# Stats Summary
# listings.describe()
# FOR LATER AFTER DATA CLEAN UP

In [9]:
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22058 entries, 0 to 22057
Data columns (total 82 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   OriginalListPrice             21968 non-null  float64
 1   ListingKey                    22058 non-null  int64  
 2   ListAgentEmail                22017 non-null  object 
 3   CloseDate                     4995 non-null   object 
 4   ClosePrice                    4377 non-null   float64
 5   ListAgentFirstName            21944 non-null  object 
 6   ListAgentLastName             22055 non-null  object 
 7   Latitude                      21759 non-null  float64
 8   Longitude                     21765 non-null  float64
 9   UnparsedAddress               22015 non-null  object 
 10  PropertyType                  22058 non-null  object 
 11  LivingArea                    19161 non-null  float64
 12  ListPrice                     21999 non-null  float64
 13  D

In [10]:
# alphabetical order, check for duplicates
listings_sorted = listings[sorted(listings.columns)]
listings_sorted.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22058 entries, 0 to 22057
Data columns (total 82 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   AboveGradeFinishedArea        0 non-null      float64
 1   AssociationFee                15514 non-null  float64
 2   AssociationFeeFrequency       6826 non-null   object 
 3   AttachedGarageYN              14359 non-null  object 
 4   BathroomsTotalInteger         20102 non-null  float64
 5   BedroomsTotal                 19230 non-null  float64
 6   BelowGradeFinishedArea        99 non-null     float64
 7   BuilderName                   729 non-null    object 
 8   BuildingAreaTotal             3531 non-null   float64
 9   BusinessType                  163 non-null    object 
 10  BuyerAgentFirstName           4798 non-null   object 
 11  BuyerAgentLastName            4812 non-null   object 
 12  BuyerAgentMlsId               4814 non-null   object 
 13  B

# Drop duplicate rows/columns

In [11]:
# Step 1: remove exact duplicates
listings = listings.loc[:, ~listings.T.duplicated()]

# Step 2: remove .1 columns (extra cleanup)
listings = listings.loc[:, ~listings.columns.str.endswith('.1')]

listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22058 entries, 0 to 22057
Data columns (total 66 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   OriginalListPrice         21968 non-null  float64
 1   ListingKey                22058 non-null  int64  
 2   ListAgentEmail            22017 non-null  object 
 3   CloseDate                 4995 non-null   object 
 4   ClosePrice                4377 non-null   float64
 5   ListAgentFirstName        21944 non-null  object 
 6   ListAgentLastName         22055 non-null  object 
 7   Latitude                  21759 non-null  float64
 8   Longitude                 21765 non-null  float64
 9   UnparsedAddress           22015 non-null  object 
 10  PropertyType              22058 non-null  object 
 11  LivingArea                19161 non-null  float64
 12  ListPrice                 21999 non-null  float64
 13  DaysOnMarket              22058 non-null  int64  
 14  ListOf

# 1. Fixing columns that has dates
```
3   CloseDate                     4995 non-null   object 
48  TaxYear                       27 non-null     float64
55  PurchaseContractDate          9248 non-null   object 
56  ListingContractDate           22058 non-null  object 
```

In [12]:
# changing it to a datetime64[ns]
date_cols = ['CloseDate', 'PurchaseContractDate', 'ListingContractDate']

for col in date_cols:
    listings[col] = pd.to_datetime(listings[col], errors='coerce')

listings[date_cols].dtypes

CloseDate               datetime64[ns]
PurchaseContractDate    datetime64[ns]
ListingContractDate     datetime64[ns]
dtype: object

In [13]:
# listings['CloseDate'].head(20)
listings['CloseDate'].tail(10)

22048   2026-03-13
22049   2026-03-13
22050          NaT
22051          NaT
22052   2026-03-18
22053   2026-03-16
22054   2026-03-10
22055   2026-03-20
22056          NaT
22057          NaT
Name: CloseDate, dtype: datetime64[ns]

In [14]:
listings['CloseDate'].isnull().sum()
# 17063 entries has a NaT out of 22058 entries

17063

In [15]:
listings['CloseDate'].isnull().mean() * 100
# about 77% has NaT

77.35515459243811

- Decided to keep the NaT for all time columns, as this can be normal for listing dates
  - Likely reason: listing has not closed yet

In [16]:
listings.tail(5)

,OriginalListPrice,ListingKey,ListAgentEmail,CloseDate,ClosePrice,ListAgentFirstName,ListAgentLastName,Latitude,Longitude,UnparsedAddress,PropertyType,LivingArea,ListPrice,DaysOnMarket,ListOfficeName,BuyerOfficeName,CoListOfficeName,ListAgentFullName,CoListAgentFirstName,CoListAgentLastName,BuyerAgentMlsId,BuyerAgentFirstName,BuyerAgentLastName,FireplacesTotal,AssociationFeeFrequency,MLSAreaMajor,TaxAnnualAmount,CountyOrParish,MlsStatus,ElementarySchool,AttachedGarageYN,ParkingTotal,BuilderName,PropertySubType,LotSizeAcres,SubdivisionName,BuyerOfficeAOR,YearBuilt,StreetNumberNumeric,ListingId,BathroomsTotalInteger,City,TaxYear,BuildingAreaTotal,BedroomsTotal,ContractStatusChangeDate,CoBuyerAgentFirstName,PurchaseContractDate,ListingContractDate,BelowGradeFinishedArea,BusinessType,StateOrProvince,MiddleOrJuniorSchool,FireplaceYN,Stories,HighSchool,Levels,LotSizeDimensions,LotSizeArea,MainLevelBedrooms,NewConstructionYN,GarageSpaces,HighSchoolDistrict,PostalCode,AssociationFee,LotSizeSquareFeet
22053,1349900.0,1152811470,Pam@TeamEtem.com,2026-03-16,1320000.0,Pamela,Etem,33.717782,-118.020937,6172 Shields,Residential,1406.0,1349900.0,14,The OC Real Estate,"Wynn Real Estate, Inc.",The OC Real Estate,Pamela Etem,Nigel,Etem,OCTRASTEV,Steven,Tran,NaN,NaN,17 - Northwest Huntington Beach,NaN,Orange,Closed,NaN,True,7.0,Prestige,SingleFamilyResidence,0.1377,Prestige (17) Tr 6181 (PRES6),OrangeCounty,1965.0,6172.0,OC26030438,2.0,Huntington Beach,NaN,NaN,3.0,2026-03-16,NaN,2026-02-24,2026-02-09,NaN,NaN,CA,NaN,True,1.0,Ocean View,One,60 x 100,6000.00,3.0,False,2.0,Huntington Beach Union High,92647,0.0,6000.0
22054,3100.0,1152811444,sarainrealestate@yahoo.com,2026-03-10,3100.0,Sara,Ramirez,34.118764,-118.156722,600 Meridian Avenue,ResidentialLease,969.0,3100.0,24,M&M Bryant Real Estate Inc,M&M Bryant Real Estate Inc,NaN,Sara Ramirez,NaN,NaN,P22275,Sara,Ramirez,NaN,NaN,658 - So. Pasadena,NaN,Los Angeles,Closed,NaN,False,2.0,NaN,SingleFamilyResidence,0.0859,Not Applicable,PasadenaFoothills,1948.0,600.0,P1-25814,1.0,South Pasadena,NaN,NaN,2.0,2026-03-10,NaN,2026-03-06,2026-02-09,NaN,NaN,CA,NaN,False,1.0,NaN,One,NaN,3741.00,NaN,NaN,2.0,NaN,91030,0.0,3741.0
22055,465000.0,1152811435,sheri@sakkrealty.com,2026-03-20,485000.0,Sheri,Jones,33.773338,-116.933692,1584 Napoli Way,Residential,1397.0,465000.0,7,S.A.K.K Realty,"HomeSmart, Evergreen Realty",S.A.K.K Realty,Sheri Jones,Karen,Belcher,PGALJAV,Javier,Galvez,NaN,NaN,SRCAR - Southwest Riverside County,NaN,Riverside,Closed,NaN,True,2.0,NaN,SingleFamilyResidence,0.2200,NaN,PacificWest,2001.0,1584.0,PTP2601082,2.0,San Jacinto,NaN,NaN,3.0,2026-03-20,NaN,2026-02-18,2026-02-10,NaN,NaN,CA,NaN,True,1.0,NaN,One,NaN,0.22,NaN,False,2.0,San Jacinto Unified,92583,0.0,9583.2
22056,599900.0,1152811419,alvinmullinsbroker@gmail.com,NaT,NaN,Alvin,Mullins,33.953730,-117.405289,4693 Central,Residential,1776.0,599900.0,27,Your Home Team Real Estate,NaN,NaN,Alvin Mullins,NaN,NaN,NaN,NaN,NaN,NaN,NaN,252 - Riverside,NaN,Riverside,Active,NaN,False,4.0,NaN,SingleFamilyResidence,0.1400,NaN,NaN,1950.0,4693.0,RS26030176,2.0,Riverside,NaN,NaN,3.0,2026-02-09,NaN,NaT,2026-02-09,NaN,NaN,CA,NaN,True,1.0,NaN,One,NaN,6098.00,3.0,False,2.0,Riverside Unified,92506,0.0,6098.0
22057,3999999.0,1152811329,davis@davisbartels.com,NaT,NaN,Davis,Bartels,34.249337,-118.789867,550 Unique Lane,Residential,6338.0,3999999.0,34,"Pinnacle Estate Properties, In",NaN,"Pinnacle Estate Properties, Inc.",Davis Bartels,Shannon,Santo,NaN,NaN,NaN,NaN,NaN,SVW - West Simi,NaN,Ventura,Active,NaN,True,4.0,NaN,SingleFamilyResidence,1.1570,NaN,NaN,1995.0,550.0,V1-34600,6.0,Simi Valley,NaN,NaN,5.0,2026-02-01,NaN,NaT,2026-02-01,NaN,NaN,CA,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


# Dropping Certain Columns
- Decided to drop certain columns since they are too sparse to be useful and/or not pertinent to the analysis that will be made later

In [19]:
#Example: 90% of this column are NaN, not useful in analysis

#listings.MiddleOrJuniorSchool
#listings['MiddleOrJuniorSchool'].isna().sum()

In [20]:
cols_to_drop = [
    'FireplacesTotal',        # 0 values
    'TaxAnnualAmount',        # 142 values only
    'TaxYear',                # 27 values only
    'BelowGradeFinishedArea', # 99 values
    'BusinessType',           # 163 values
    'BuilderName',            # 729 values
    'LotSizeDimensions',      # 1151 values
    'ElementarySchool',
    'MiddleOrJuniorSchool',
    'HighSchool'
]

listings_col_dropped = listings.drop(columns=cols_to_drop)

In [25]:
listings_col_dropped.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22058 entries, 0 to 22057
Data columns (total 59 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   OriginalListPrice         21968 non-null  float64       
 1   ListingKey                22058 non-null  int64         
 2   ListAgentEmail            22017 non-null  object        
 3   CloseDate                 4995 non-null   datetime64[ns]
 4   ClosePrice                4377 non-null   float64       
 5   ListAgentFirstName        21944 non-null  object        
 6   ListAgentLastName         22055 non-null  object        
 7   Latitude                  21759 non-null  float64       
 8   Longitude                 21765 non-null  float64       
 9   UnparsedAddress           22015 non-null  object        
 10  PropertyType              22058 non-null  object        
 11  LivingArea                19161 non-null  float64       
 12  ListPrice         

In [26]:
# Price per sqft
listings_col_dropped['PricePerSqft'] = listings_col_dropped['ListPrice'] / listings_col_dropped['LivingArea']

In [27]:
# Time Features
listings_col_dropped['Year'] = listings_col_dropped['ListingContractDate'].dt.year
listings_col_dropped['Month'] = listings_col_dropped['ListingContractDate'].dt.month

In [32]:
# Decided to drop a lot more columns that's not pertinent to the analysis and only keep the important ones
important_cols = [
    'ListingKey',
    'ListPrice',
    'ClosePrice',
    'LivingArea',
    'LotSizeArea',
    'BedroomsTotal',
    'BathroomsTotalInteger',
    'City',
    'CountyOrParish',
    'PropertyType',
    'PropertySubType',
    'MlsStatus',
    'DaysOnMarket',
    'YearBuilt',
    'CloseDate',
    'ListingContractDate',
    'PurchaseContractDate',
    'Latitude',
    'Longitude'
]

listings_clean = listings_col_dropped[important_cols]

In [31]:
listings_clean.describe

,ListingKey,ListPrice,ClosePrice,LivingArea,LotSizeArea,BedroomsTotal,BathroomsTotalInteger,DaysOnMarket,YearBuilt,Latitude,Longitude
count,22058,21999,4377,19161,20208,19230,20102,22058,20175,21759,21765
mean,1153419696,1027269,813386,1927,98540,3,3,16,1979,35,-118
std,530056,2169992,1116016,1623,4207138,2,2,14,28,2,4
min,1152811329,1,425,0,0,0,0,-15,1776,0,-124
25%,1153075484,169999,4400,1175,4967,2,2,5,1960,34,-119
50%,1153313298,654999,579000,1609,7372,3,2,13,1979,34,-118
75%,1153590072,1198919,1200000,2260,14625,4,3,26,2001,34,-117
max,1156277542,100000000,17600000,94619,418742280,149,149,407,2027,42,118


# Initial Analysis
- Listing price: good, but max looks MASSIVE
- Living Area: ok
- Bedrooms/Bathrooms: dist looks good, but is there really a home with 149 bed/bathrooms?
- DaysOnMarket: bad, why is it negative? We can't go back in time
- LivingArea: 94619 sqft....that HUGE, I guess it works out if there is a home with 149 bed & bathrooms lol
- LotSizeArea: 418742280 acre?!?! That roughly the size of ALASKA!

Possible indicator of major outliers

# Key Takeaway
The dataset shows a realistic distribution of housing characteristics, but contains extreme outliers such as unusually high bedroom counts, lot sizes, and negative days on market, which I would clean before further analysis.